js/apps/events-page/controllers/eventsPageCtrl.js

```diff
@@ -208,23 +249,126 @@
 			$collapseIcon.toggleClass('fa-plus-square').toggleClass('fa-minus-square');
 		};
 
-		/* ** Init ** */
+		const getKeywords = () => {
+			return $location.search().term && $location.search().term.length ? 
+				$location.search().term :
+				"";
+		};
 
+		/* ** Init ** */
 		angular.element(document).on('hide.bs.collapse', '.expando-wrapper .collapse', toggleIcon);
 		angular.element(document).on('show.bs.collapse', '.expando-wrapper .collapse', toggleIcon);
 
-		if ($location.search().term && $location.search().term.length) {
-			self.keywords = $location.search().term;
-			self.keywordSearch();
-		} else {
-			eventsService
-				.get(requestModel)
-				.then(processEvents)
-				.catch(handleFailedEventsGetRequest);
-		}
+		const getFilterId = (filterType, val) => {
+			if (!val) return;
+
+			if (Object.hasOwnProperty.call(vm.data, filterType)) {
```

Comment:
Should this be `Object.prototype.hasOwnProperty.call(...)`?

Reply:
fixed, you are right on this one